src/components/Map.js

```diff
@@ -0,0 +1,62 @@
+import GoogleMap from 'google-map-react'
+import PropTypes from 'prop-types'
+import React from 'react'
+
+import Cluster from './Cluster'
+import Location from './Location'
+
+const Map = (props) => {
+  const {
+    bootstrapURLKeys,
+    view,
+    markerData,
+    showLocations,
+    onZoomAnimationEnd,
+    onDragEnd,
+    onClusterClick,
+    onLocationClick,
+  } = props
+
+  return (
+    <GoogleMap
+      bootstrapURLKeys={bootstrapURLKeys}
+      center={view.center}
+      zoom={view.zoom}
+      onZoomAnimationEnd={onZoomAnimationEnd}
+      onDragEnd={onDragEnd}
+    >
+      {markerData.map((marker, index) =>
+        showLocations ? (
+          <Location
+            key={marker.id}
+            onClick={onLocationClick}
```

Comment:
In order to solve the issue of not knowing which location or cluster was clicked, I'd suggest that you make an anonymous function here that calls `onLocationClick` or `onClusterClick` with the entire location or cluster object, respectively.

In terms of data flow, it doesn't make sense to me that the UI components `Cluster` and `Location` should know about their individual properties (other than count, because that affects the diameter of the circle), since higher-level components can & should take care of that.